src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/SubAtlasCommand.java

```diff
@@ -148,19 +178,119 @@ protected void processAtlas(final Atlas atlas, final String atlasFileName,
-                .getOptionArgument(CUT_TYPE_OPTION_LONG).orElse("SOFT_CUT");
+        final Optional<String> cutTypeParameter = this.optionAndArgumentDelegate
+                .getOptionArgument(CUT_TYPE_OPTION_LONG);
+        final Optional<String> wktParameter = this.optionAndArgumentDelegate
+                .getOptionArgument(POLYGON_OPTION_LONG);
+        final Optional<String> predicateParameter = this.optionAndArgumentDelegate
+                .getOptionArgument(PREDICATE_OPTION_LONG);
 
-        try
+        if (cutTypeParameter.isPresent())
         {
-            this.cutType = AtlasCutType.valueOf(cutTypeString.toUpperCase());
+            try
+            {
+                this.cutType = AtlasCutType.valueOf(cutTypeParameter.get().toUpperCase());
+            }
+            catch (final IllegalArgumentException exception)
+            {
+                this.outputDelegate
+                        .printlnErrorMessage("invalid cut type " + cutTypeParameter.get());
+                this.outputDelegate
+                        .printlnStderr("Try " + new StringList(CUT_TYPE_STRINGS).join(", "));
+                return 1;
+            }
         }
-        catch (final IllegalArgumentException exception)
+
+        if (wktParameter.isPresent())
         {
-            this.outputDelegate.printlnErrorMessage("invalid cut type " + cutTypeString);
-            this.outputDelegate.printlnStderr("Try " + new StringList(CUT_TYPE_STRINGS).join(", "));
-            return 1;
+            final WKTReader reader = new WKTReader();
+            Geometry geometry = null;
+            try
+            {
+                geometry = reader.read(wktParameter.get());
+            }
+            catch (final ParseException exception)
+            {
+                logger.error("unable to parse {}", wktParameter.get(), exception);
+                return 1;
+            }
+
+            if (geometry instanceof org.locationtech.jts.geom.Polygon)
+            {
+                this.polygon = Optional.ofNullable(new JtsPolygonConverter()
+                        .backwardConvert((org.locationtech.jts.geom.Polygon) geometry));
+            }
+            else
+            {
+                this.outputDelegate
+                        .printlnErrorMessage("unsupported geometry type " + wktParameter.get());
+                return 1;
+            }
         }
+
+        if (predicateParameter.isPresent())
+        {
+            final Optional<Predicate<AtlasEntity>> predicate = getPredicateFromCommandLineExpression(
+                    predicateParameter);
+            if (!predicate.isPresent())
+            {
+                this.outputDelegate.printlnErrorMessage("could not parse predicate");
+                return 1;
+            }
+            this.matcher = predicate;
+        }
+
         return 0;
     }
+
+    private Optional<Predicate<AtlasEntity>> getPredicateFromCommandLineExpression(
```

Comment:
Can this be broken out in a separate converter?

Reply:
It might be useful in other contexts!

Reply:
This is a good idea.